diff --git a/vpp-api/python/pneum/pneum.c b/vpp-api/python/pneum/pneum.c
index ebe47b2..f6f3d78 100644
--- a/vpp-api/python/pneum/pneum.c
+++ b/vpp-api/python/pneum/pneum.c
@@ -147,6 +147,28 @@ pneum_connect (char * name, char * chroot_prefix)
   return (0);
 }
 
+
+int
+pneum_connect_sync (char *name)
+{
+  int rv = 0;
+  pneum_main_t *pm = &pneum_main;
+
+  if ((rv = vl_client_api_map("/vpe-api"))) {
+    clib_warning ("vl_client_api map rv %d", rv);
+    return rv;
+  }
+
+  if (vl_client_connect(name, 0, 32) < 0) {
+    vl_client_api_unmap();
+    return (-1);
+  }
+
+  pm->connected_to_vlib = 1;
+
+  return (0);
+}
+
 int
 pneum_disconnect (void)
 {
@@ -198,6 +220,12 @@ pneum_read (char **p, int *l)
   return (rv);
 }
 
+void
+pneum_data_free(char *data)
+{
+  vl_msg_api_free(data);
+}
+
 /*
  * XXX: Makes the assumption that client_index is the first member
  */
diff --git a/vpp-api/python/pneum/pneum.h b/vpp-api/python/pneum/pneum.h
index 75b10f8..7ca66ff 100644
--- a/vpp-api/python/pneum/pneum.h
+++ b/vpp-api/python/pneum/pneum.h
@@ -20,4 +20,10 @@ int pneum_disconnect(void);
 int pneum_read(char **data, int *l);
 int pneum_write(char *data, int len);
 
+/* Synchronous API connect - no callback */
+int pneum_connect_sync(char *name);
+
+/* Call to free the data read via pneum_read */
+void pneum_data_free(char *data);
+
 #endif
