diff --git a/vpp-api/python/pneum/pneum.c b/vpp-api/python/pneum/pneum.c
index 7250cf6..9f4087f 100644
--- a/vpp-api/python/pneum/pneum.c
+++ b/vpp-api/python/pneum/pneum.c
@@ -144,6 +144,28 @@ pneum_connect (char *name)
   return (0);
 }
 
+
+int
+pneum_connect_sync (char *name)
+{
+  int rv = 0;
+  pneum_main_t *pm = &pneum_main;
+
+  if ((rv = vl_client_api_map("/vpe-api"))) {
+    clib_warning ("vl_client_api map rv %d", rv);
+    return rv;
+  }
+
+  if (vl_client_connect(name, 0, 32) < 0) {
+    vl_client_api_unmap();
+    return (-1);
+  }
+
+  pm->connected_to_vlib = 1;
+
+  return (0);
+}
+
 int
 pneum_disconnect (void)
 {
@@ -195,6 +217,12 @@ pneum_read (char **p, int *l)
   return (rv);
 }
 
+void
+pneum_data_free(char *data)
+{
+  vl_msg_api_free(data);
+}
+
 /*
  * XXX: Makes the assumption that client_index is the first member
  */
diff --git a/vpp-api/python/pneum/pneum.h b/vpp-api/python/pneum/pneum.h
index 75fccf8..75f87d1 100644
--- a/vpp-api/python/pneum/pneum.h
+++ b/vpp-api/python/pneum/pneum.h
@@ -16,8 +16,10 @@
 #define included_pneum_h
 
 int pneum_connect(char *name);
+int pneum_connect_sync(char *name);
 int pneum_disconnect(void);
 int pneum_read(char **data, int *l);
+void pneum_data_free(char *data);
 int pneum_write(char *data, int len);
 
 #endif
